tests/framework/microvm.py

```diff
@@ -355,6 +356,43 @@ def init_snapshot_api(self):
             self._api_session
         )
 
+    @property
+    def vcpus_count(self):
+        """Get the vcpus count."""
+        return self._vcpus_count
+
+    @vcpus_count.setter
+    def vcpus_count(self, vcpus_count: int):
+        """Set the vcpus count."""
+        self._vcpus_count = vcpus_count
+
+    def pin_vmm(self, cpu_ids_list: list) -> bool:
+        """Pin the firecracker process VMM thread to a cpu list."""
+        if self.jailer_clone_pid:
+            for thread in utils.ProcessManager.get_threads(
+                    self.jailer_clone_pid)["firecracker"]:
+                utils.ProcessManager.set_cpu_affinity(thread, cpu_ids_list)
+                return True
+        return False
+
+    def pin_vcpu(self, vcpu_id: int, cpu_ids_list: list):
```

Comment:
why do we need a list of cpu_ids ? Would it make sense to pin a single vCPU to multiple CPUs ?

Reply:
I can not tell if there is a specific usecase right now, but we might want to set the affinity for different vCPU threads to different cpu lists, pertaining to different NUMA nodes, not necessarily to a single core. I think we should see this API as support for process cpu affinity modifications rather than strict process/cpu pinning. We can restrict the support for now or rename the API to suggest better that it is intended to be used for setting cpu affinity for microvm threads, as in "set_vcpu_thread_cpu_affinity`. WDYT?

Reply:
Personally I would restrict the support for now. I would only offer the ability to pin a vCPU to 1 cpu_id. The same for vmm, api, etc.

Reply:
Could we just pass a mask here? (e.g. `0x3`)

It would work for both one physical core and multiple cores.

Reply:
Done.